code/modules/events/vox_trade_probe.dm

```diff
@@ -0,0 +1,116 @@
+/*
+ * Vox Trade Probe - if no trader, trade shuttle in dock, probe has not fired yet, wages are not below 100%, port is open
+ * Fire weight can increase up to 90, which is very high for an event, if lot of players, money, and time
+ */
+
+ #define TRADEPROBE_BONUS_TIMER 120 //how many seconds need to pass to increase weight by 1
+
+/datum/event/tradeprobe
+	endWhen	= 270 //How many 2-second ticks should we stay? 9 minutes = 540 / 2 = 270
+	oneShot = TRUE //one per shift
+
+/datum/event/tradeprobe/can_start(var/list/active_with_role)
+	//No probe if... there is a trader, the trade shuttle is at the station, or financial mismanagement
+	if(SStrade.loyal_customers.len)
+		return FALSE
+	if(!istype(trade_shuttle.current_port,/obj/docking_port/destination/trade/start))
```

Comment:
Should also check if a custom shuttle isn't at the trading port currently. Although the collision would be funny, I guess.